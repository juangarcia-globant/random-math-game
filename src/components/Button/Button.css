
.button {
  background: #61dafb;
  padding: 10px 20px;
  border: 1px solid #057897;
  border-radius: 5px;
  color: white;
  font-size: 1em;
  font-weight: 600;
  cursor: pointer;
}
