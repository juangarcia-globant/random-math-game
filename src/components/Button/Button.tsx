import './Button.css';

type ButtonProps = {
  text: string,
  onClick?: () => void,
};

export const Button: React.FunctionComponent<ButtonProps> = ({
  text = 'button',
  onClick,
}) => {
  return (
    <button className="button" onClick={onClick}>
      {text}
    </button>
  );
}

export default Button;
