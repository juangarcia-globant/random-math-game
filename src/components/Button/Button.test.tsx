import { render, screen, fireEvent } from '@testing-library/react'
import Button from './Button';

describe("<Button />", () => {
  test("Matches the snapshot", () => {
    const { asFragment } = render(<Button text="Test" />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("Button should be render", () => {
    render(<Button text="Test" />);
    expect(screen.getByRole('button', { name: "Test" })).toBeInTheDocument();
  });

  test("The function should be executed on click", async () => {
    const func = jest.fn();
    render(<Button text="Test" onClick={func} />);
    fireEvent.click(screen.getByRole('button', { name: "Test" }));
    expect(func).toHaveBeenCalledTimes(1);
  });
});