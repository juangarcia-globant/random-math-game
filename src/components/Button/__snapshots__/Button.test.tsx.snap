// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Matches the snapshot 1`] = `
<DocumentFragment>
  <button
    class="button"
  >
    Test
  </button>
</DocumentFragment>
`;
